core/src/main/java/com/sequenceiq/cloudbreak/service/parcel/ParcelFilterService.java

```diff
@@ -39,26 +39,30 @@
     @Inject
     private RestClientFactory restClientFactory;
 
-    public Set<ClouderaManagerProduct> filterParcelsByBlueprint(Set<ClouderaManagerProduct> parcels, Blueprint blueprint) {
+    @Inject
+    private ParcelInfoProvider parcelInfoProvider;
+
+    public Set<ClouderaManagerProduct> filterParcelsByBlueprint(Long stackId, Set<ClouderaManagerProduct> parcels, Blueprint blueprint) {
+        LOGGER.debug("Filtering the following parcels based on the blueprint {}", parcels);
         Set<String> serviceNamesInBlueprint = getAllServiceNameInBlueprint(blueprint);
         LOGGER.debug("The following services are found in the blueprint: {}", serviceNamesInBlueprint);
         Set<ClouderaManagerProduct> ret = new HashSet<>();
         if (serviceNamesInBlueprint.contains(null)) {
             LOGGER.debug("We can not identify one of the service from the blueprint so to stay on the safe side we will add every parcel");
             return parcels;
         }
-        filterParcels(parcels, serviceNamesInBlueprint, ret);
+        filterParcels(stackId, parcels, serviceNamesInBlueprint, ret);
         LOGGER.debug("The following parcels are used in CM based on blueprint: {}", ret);
         return ret;
     }
 
-    private void filterParcels(Set<ClouderaManagerProduct> parcels, Set<String> serviceNamesInBlueprint, Set<ClouderaManagerProduct> ret) {
+    private void filterParcels(Long stackId, Set<ClouderaManagerProduct> parcels, Set<String> serviceNamesInBlueprint, Set<ClouderaManagerProduct> ret) {
         parcels.forEach(parcel -> {
             ImmutablePair<ManifestStatus, Manifest> manifest = readRepoManifest(parcel.getParcel());
             if (manifest.right != null && ManifestStatus.SUCCESS.equals(manifest.left)) {
                 Set<String> componentNamesInParcel = getAllComponentNameInParcel(manifest.right);
                 LOGGER.debug("The following components are available in parcel: {}", componentNamesInParcel);
-                if (componentNamesInParcel.stream().anyMatch(serviceNamesInBlueprint::contains)) {
+                if (componentNamesInParcel.stream().anyMatch(serviceNamesInBlueprint::contains) || isCustomParcel(stackId, parcel)) {
```

Comment:
the log below only refers to the first if clause, pls create a separate log entry if `isCustomParcel` is true to be able to debug easily

Reply:
agree, the anyomatch part could be also in a method with log